source/crud/tests/write/findOneAndUpdate-arrayFilters.yml

```diff
@@ -11,16 +11,16 @@ tests:
             arguments:
                 filter: {}
                 update:
-                    $set: {y.$[i].b: 2}
+                    $set: {"y.$[i].b": 2}
```

Comment:
The Ruby yaml parser is unable to parse array braces as part of a key without it being quoted. I suspect other drivers that parse the yaml rather than the JSON might run into this issue as well.